client/components/add-new-site/modals/index.tsx

```diff
@@ -0,0 +1,19 @@
+import { useMemo } from 'react';
+import AsyncLoad from 'calypso/components/async-load';
+import isA8CForAgencies from 'calypso/lib/a8c-for-agencies/is-a8c-for-agencies';
+
+const AddNewSiteModals = () => {
+	const renderContent = useMemo( () => {
+		switch ( true ) {
+			case isA8CForAgencies():
+				return (
+					<AsyncLoad require="calypso/components/add-new-site/modals/a4a" placeholder={ null } />
+				);
+			default:
+				return null;
+		}
+	}, [] );
+	return renderContent;
```

Comment:
As mentioned above, I'd skip the memoization here and would just go with an `if` and early return.

Reply:
Makes sense. Changed it [here](https://github.com/Automattic/wp-calypso/pull/97499/commits/8365a989faeeda3c1d128f37813e2595c66f08de).